src/transform/predicate_pushdown.rs

```diff
@@ -74,6 +74,38 @@ impl crate::Transform for PredicatePushdown {
 }
 
 impl PredicatePushdown {
+    fn is_safe_leaf(s: &ScalarExpr, input_arity: usize) -> bool {
+        match s {
```

Comment:
As a style thing (and probably folks disagree) I've been writing the code in the module as top-down with entry points first and helpers second. We don't have to do that, but if helpers come first probably best to put some comments on methods explaining what they do and are for (vs "don't try to read this until you've read the rest of the file").